### ISSUE TITLE
TDML Data Source Deserialization Error in 2.18.1

### ISSUE THREAD

**Issue by jm98** [TabularEditor/TabularEditor#1092]
Steps:

1. Create new model with TDML serialization
2. Create LEGACY data source using Tabular Editor's UI
    - name:  DS1
    - account: DOMAIN/user1234
    - connectionString: "Data source=(local);Initial Catalog=MyDatabase;Integrated Security=true"
    - impersonationMode: ImpersonateAccount
    - provider: System.Data.SqlClient
3. Save as Folder...

Observation:
**datasources.tmd**
```
dataSource DS1
    account: DOMAIN/user1234
    connectionString: "Data source=(local);Initial Catalog=MyDatabase;Integrated Security=true"
    impersonationMode: ImpersonateAccount
    provider: System.Data.SqlClient
```

4. Close Tabular Editor and restart
5. Load model

Get:
```
Error loading Model from disk
KeyNotFoundException encountered while deserializing
TDML: account
```

**NOTE**
If you remove the account impersonation (`account` & `impersonationMode` properties) then the deserialization fails on `connectionString`

**Comment by mthierba** [comment]
That's a known limitation of the 1st preview of TMDL: https://learn.microsoft.com/analysis-services/tmdl/tmdl-overview?view=asallproducts-allversions#considerations-and-limitations
It will be resolved in an upcoming release.
Thank you for reporting as that helps filling those gaps!

**Comment by jm98** [comment]
Thanks @mthierba.  Is there a work around or do I have to abandon TDML until the deserialization is supported?  (I hope for the 1st option)

**Comment by mthierba** [comment]
The only viable workaround for the time being would be to remove that datasource from the model. And only add it back in during deployment, for instance. I appreciate that's not ideal.

**Comment by otykier** [comment]
> it does leave “a bad taste”

which is why this feature is marked as “preview” :-)

**Comment by mthierba** [comment]
Pleased to report that the underlying issue has been fixed internally - this will become available to everyone with the next TOM release (those are usually monthly).

**Comment by jm98** [comment]
Thank you.

**Comment by otykier** [comment]
Fixed in [2.20.0](https://github.com/TabularEditor/TabularEditor/releases/tag/2.20.0). Thanks for reporting!
